### ISSUE TITLE
add includes to target

### ISSUE THREAD

**Issue by jakkes** [RustingSword/tensorboard_logger#15]
I use this repo as a dependency, however I had to alter the CMake file slightly. 

From
```
include_directories(
    ${TENSORBOARD_INCLUDE_PATH}
    ${Protobuf_INCLUDE_DIRS}
)

add_library(tensorboard_logger STATIC
  ${sources}
  ${PROTO_SRCS}
)
```

To
```
add_library(tensorboard_logger STATIC
  ${sources}
  ${PROTO_SRCS}
)

target_include_directories(
  tensorboard_logger PRIVATE ${Protobuf_INCLUDE_DIRS}
)
target_include_directories(
  tensorboard_logger PUBLIC ${TENSORBOARD_INCLUDE_PATH}
)
```

This allows adding the header files by simply linking the library to whatever library I am working on, instead of manually linking the header directory. Could this be added to master?

**Comment by RustingSword** [comment]
Sure, feel free to send a PR so I can merge it. However I just tested your proposition, and compilation of this project itself failed due to `fatal error: 'google/protobuf/port_def.inc' file not found`. If I change `PRIVATE` to `PUBLIC`, it successfully compiles. I'm not quite sure what's the correct way to do this.

**Comment by jakkes** [comment]
Ok, I see. I'll give it a try.

**Comment by mcourteaux** [comment]
This can be closed now, by merging #16 .
